lms/djangoapps/grades/tasks.py

```diff
@@ -146,12 +155,43 @@ def _has_database_updated_with_new_score(
         )
         if api_score is None:
             # Same case as the initial 'if' above, for submissions-specific scores
+            if not score_deleted:
+                _log_db_not_updated(
+                    user_id,
+                    scored_block_usage_key,
+                    expected_modified_time,
+                    score_deleted,
+                    "N/A, no score present in database."
+                )
             return score_deleted
         reported_modified_time = api_score['created_at']
     else:
         reported_modified_time = score.modified
 
-    return reported_modified_time >= expected_modified_time
+    db_updated = reported_modified_time >= expected_modified_time
+    if not db_updated:
+        _log_db_not_updated(
+            user_id,
+            scored_block_usage_key,
+            expected_modified_time,
+            score_deleted,
+            reported_modified_time
+        )
+    return db_updated
+
+
+def _log_db_not_updated(user_id, scored_block_usage_key, expected_modified_time, score_deleted, modified_time):
+    log.info(
+        u"Persistent Grades: tasks._has_database_updated_with_new_score is False. User ID: {0}. Block usage key: {1}. "
+        u"Expected modified time: {2}. Score deleted: {3}. Actual modified: {4}. Task ID: {5}.".format(
+            user_id,
```

Comment:
Can we pass all the kwargs passed to the task and simply log those as a dict value?  That would be most forward compatible.